network/src/protocols/discovery.rs

```diff
@@ -55,17 +61,17 @@ impl ServiceProtocol for DiscoveryProtocol {
         let discovery_task = self
             .discovery
             .take()
-            .map(|discovery| {
+            .map(|mut discovery| {
                 debug!("Start discovery future_task");
-                discovery
-                    .for_each(|()| Ok(()))
-                    .map_err(|err| {
-                        warn!("discovery stream error: {:?}", err);
-                    })
-                    .then(|_| {
-                        debug!("End of discovery");
-                        Ok(())
-                    })
+                async move {
+                    loop {
```

Comment:
Any reason that we have to replace all the `for_each`s with an async loop?

Reply:
This is a question of writing style, the two are equivalent